protocol/packages/currencies/build/main.rs

```diff
@@ -0,0 +1,148 @@
+use std::{
+    env,
+    fs::File,
+    io::{self, Write},
+    path::{Path, PathBuf},
+};
+
+use anyhow::{bail, Context as _, Result};
+
+use topology::{CurrencyDefinitions, Topology};
+
+use self::{currencies_tree::CurrenciesTree, either::Either, protocol::Protocol};
+
+mod convert_case;
+mod currencies_tree;
+mod either;
+mod protocol;
+mod sources;
+mod subtype_lifetime;
+
+const PROTOCOL_JSON: &str = "../../../build-configuration/protocol.json";
+
+const TOPOLOGY_JSON: &str = "../../../build-configuration/topology.json";
+
+const BUILD_REPORT: &str = "CURRENCIES_BUILD_REPORT";
+
+fn main() -> Result<()> {
+    for path in ["build/", PROTOCOL_JSON, TOPOLOGY_JSON] {
+        println!("cargo::rerun-if-changed={path}");
+    }
+
+    println!("cargo::rerun-if-env-changed={BUILD_REPORT}");
+
+    let output_directory: &Path = &PathBuf::from(
+        env::var_os("OUT_DIR").context("Cargo did not set `OUT_DIR` environment variable!")?,
+    );
+
+    let files_exist = IntoIterator::into_iter([PROTOCOL_JSON, TOPOLOGY_JSON])
```

Comment:
please rename the variable to become more descriptive, for example, `definition_found`